src/test/java/picard/sam/RevertSamTest.java

```diff
@@ -445,4 +443,14 @@ public void testGetDefaultExtension() {
         Assert.assertEquals(RevertSam.getDefaultExtension("this.is.a.bam"), ".bam");
         Assert.assertEquals(RevertSam.getDefaultExtension("foo"), ".bam");
     }
+
+    @Test(expectedExceptions = PicardException.class)
+    public void testNoRgInfo() {
```

Comment:
can you make sure that running without OUTPUT_BY_READGROUP works in these cases?